### ISSUE TITLE
Linker error: undefined reference to `yarp::dev::PolyDriver::getImplementation()'

### ISSUE THREAD

**Issue by PeterBowman** [roboticslab-uc3m/asibot-hmi#2]
Failure reported on YARP 2.3.68.1 and current master (**but** no errors on Travis builds both on master and devel):

```
~/roboticslab/asibot-hmi/build (develop u=)$ make
[ 25%] Linking CXX executable ../../bin/webInterface
CMakeFiles/webInterface.dir/WebResponder.cpp.o: In function `roboticslab::WebResponder::read(yarp::os::ConnectionReader&)':
WebResponder.cpp:(.text+0x5ea4): undefined reference to `yarp::dev::PolyDriver::getImplementation()'
collect2: error: ld returned 1 exit status
programs/webInterface/CMakeFiles/webInterface.dir/build.make:152: recipe for target 'bin/webInterface' failed
make[2]: *** [bin/webInterface] Error 1
CMakeFiles/Makefile2:203: recipe for target 'programs/webInterface/CMakeFiles/webInterface.dir/all' failed
make[1]: *** [programs/webInterface/CMakeFiles/webInterface.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by PeterBowman** [comment]
I'm not able to replicate this on the `web-interface` branch (Ubuntu Xenial on WSL, YARP 2.3.70.2). @jgvictores could you please give it a try?

**Comment by PeterBowman** [comment]
Note: compiled successfully on 822cd8e (`web-interface`), but no luck on its parent commit.

**Comment by jgvictores** [comment]
- `develop` ATOW (7fd2b5661d4871673526d9413af46ef50169b351): linker error
- Previous `develop` ATOW (1d1015178e3d9c5b19ed6a1fee061f95dd041e2c): linker error
- `web-interface` ATOW (df1f9e299ef4bd6924afbdc50004d985f4d38741): compiles
- Previous `web-interface` ATOW (a37047d166805b780889e5b5b4ea27c14be043e5): compiles

**Comment by jgvictores** [comment]
Sometimes strange things happen depending on compiler/linker order. Maybe just merge `web-interface` to `develop` and be happy with issue solved?

**Comment by PeterBowman** [comment]
Happiness is hard to achieve. I need to know first why I am to be happy.

Interesting:

* CMake generates the same build files (`webInterface.dir` folder) both on 822cd8e (works) and 19379ac (fails, parent commit).
* I've always worked with this repo on Ubuntu, no doubts here. But I hadn't upgrade YARP for a long time, that's why I may have missed this issue. On `develop`, YARP v2.3.68.1+ makes it crash, but the app is successfully linked against YARP v2.3.66.2.

**Comment by PeterBowman** [comment]
No linker errors on YARP v2.3.71 (current `devel`). Perhaps compiler/linker flags differ between YARP itself and `asibot-hmi`, see https://stackoverflow.com/a/19069107.

**Comment by PeterBowman** [comment]
I went the 'I don't know what I'm doing' way and merged `web-interface` into `develop` (211234d). Problem "solved", but comments are welcome in case anyone knows what could have happened here.
